cwl/src/main/scala/cwl/CwlWomExpression.scala

```diff
@@ -155,8 +166,22 @@ final case class InitialWorkDirFileGeneratorExpression(entry: IwdrListingArrayEn
             writtenFile <- Try(Await.result(ioFunctionSet.writeFile(entryname, contentString), Duration.Inf)).toErrorOr
           }  yield writtenFile
         }
+      case IwdrListingArrayEntry.ECMAScriptExpression(expression) =>
+        // A single expression which must evaluate to an array of Files
+        val expressionEvaluation: ErrorOr[WomValue] = ExpressionEvaluator.evalExpression(expression, ParameterContext().withInputs(inputValues, ioFunctionSet)).toErrorOr
+
+        expressionEvaluation flatMap {
+          case array: WomArray if WomArrayType(WomSingleFileType).coercionDefined(array) =>
+            val newFileArray: ErrorOr[List[WomFile]] = array.value.map(_.valueString).toList.traverse { source: String =>
+              val dest = source.split('/').last
+              Try(Await.result(ioFunctionSet.copyFile(source, dest), Duration.Inf)).toErrorOr
```

Comment:
ToL The 5 `Await.result(..., Duration.Inf)`s in this file suggest to me these functions maybe shouldn't be synchronous... 🙂

Reply:
This does look to be very similar to the yuck above.

Reply:
I agree, but this is probably a bigger problem than just `InitialWorkDirRequirement` (this is itself copy/paste from elsewhere)